src/ray/core_worker/object_interface.cc

```diff
@@ -3,18 +3,38 @@
 #include "ray/common/ray_config.h"
 #include "ray/core_worker/object_interface.h"
 #include "ray/core_worker/store_provider/local_plasma_provider.h"
+#include "ray/core_worker/store_provider/memory_store_provider.h"
 #include "ray/core_worker/store_provider/plasma_store_provider.h"
 
 namespace ray {
 
+void GetObjectIdsPerStoreProvider(
+    const std::vector<ObjectID> &object_ids,
+    EnumUnorderedMap<StoreProviderType, std::unordered_set<ObjectID>> *results) {
+  for (const auto &object_id : object_ids) {
+    // By default use `PLASMA` store provider.
+    auto type = StoreProviderType::PLASMA;
+    if (object_id.IsReturnObject() &&
+        object_id.GetTransportType() ==
+            static_cast<int>(TaskTransportType::DIRECT_ACTOR)) {
+      // Direct actor call uses `MEMORY` store provider.
```

Comment:
consolidate this into the comment above that explains the `IsReturnObject()` check

Reply:
sure.